src/rules/russian.toml

```diff
@@ -0,0 +1,118 @@
+# (for example, opening quotes without closing)
+disallowed_symbols = [
+  '<', '>', '+', '*', '\', '#', '@', '^', '[', ']', '(', ')', '/',
+  'é', 'è', 'à', 'ç', 'Å', '薛', '⋅', '·', 'ѿ', '=', '|', '≡', '→',
+  '×', 'Ա', 'բ', 'դ', 'ի', 'ս', 'յ', 'շ', 'ո', '•', 'थ्', '©', '¸',
+  '天', '不', '言', '，', '以', '行', '與', '事', '示', '之', '而',
+  '已', '矣', '《', '大', '学', '集', '注', '》', '多', '识', '录',
+  '「', '鴃', '」', '鳩', '鳧', '鶯', 'ا', 'ل', 'ر', 'د', 'ن',
+  'เ', 'วฺ', 'ส', 'ลฺ', 'ยุ', 'ทิ', 'โ', 'ร', 'ชฺ', 'ท', 'ยุ', 'ตฺ', 'สฺ',
+  'ย', 'พ', 'ทฺ', 'นิ', 'इ', '_', 'Ѳ', '‹', '›', '`',
+  'і', 'ў', '~', '®', '%', '⟨', '⟩', '¬', '&', '≤',
+  'α', 'β', 'Γ', 'γ', 'Δ', 'δ', 'ε', 'ζ', 'η', 'Θ', 'θ', 'ι', 'Ә',
+  'Λ', 'λ', 'μ', 'ν', 'Ξ', 'ξ', 'ρ', 'Σ', 'σ', 'ς', 'τ',
+  'υ', 'Φ', 'φ', 'χ', 'Ψ', 'ψ', 'Ω', 'ω', 'a', 'A', 'b', 'B', 'c',
+  'C', 'd', 'D', 'e', 'E', 'f', 'F', 'g', 'G', 'h', 'H', 'i', 'I',
+  'j', 'J', 'k', 'K', 'l', 'L', 'm', 'M', 'n', 'N', 'o', 'O', 'p',
+  'P', 'q', 'Q', 'r', 'R', 's', 'S', 't', 'T', 'u', 'U', 'v', 'V',
+  'w', 'W', 'x', 'X', 'y', 'Y', 'z', 'Z',
+  'җ', 'ә', 'ү', 'ː', 'қ', 'ћ', 'Ä', 'Ҋ', 'ј', 'Ι', 'Χ',
+  
+  # There is a lot of broken sentences like:
+  # У меня щемило сердце: «Что мы сделаем?
+  # Такой он был веселый, бодрый, нас воодушевлял: „Вы знаете, не бойтесь!
+  # История и культура»".
+  # Пушкин, прочитав поэму, сказал «— Далеко мальчик пойдет».
+  # После увольнения он пытается создать новую жизнь...Жизнь, в свою очередь, строит ему множество «баррикад"
+  # Пушкин, Набоков… мало кого еще можно поставить в этот ряд?
+  # Let's remove them all, we have so many good sentences without problems
+  '«', '»', '"', '„', '“', '”', '‘', '…',
+  '{', '}', '（', '）',
+
+  # Sentences with accented chars, it is not possible to voice and
+  # should be removed, example:
+  # е́р
+  'ѣ', 'Ѣ', 'І', 'ң', 'ă', 'ĕ', ' ́', '́',
+  
+  # Sentences with invisible chars (use vim to work with them)
+  '­', '​', '﻿', '་'
+]
+
+broken_whitespace = ["  ", " ,", " .", " ?", " !", " ;", " \""]
+
+abbreviation_patterns = [
+  # Abbreviations:
+  
+  # 1. М.Н.С.
+  "[А-Я]+\\.*[А-Я]",
+  
+  # 2. А. Пушкин
+  "[А-Я]\\.",
+
+  # 3. Дж.
+  "[А-Я][а-я]\\.",
+  
+  # 4. СССР
+  "[А-Я]{2,}",
+
+  # 5. г. Пушкина.
+  "\\s[а-я]\\.",
+
+  # 6. — —
+  "— —",
+
+  # 7. сайка фито— и зоопланктоном
+  "[а-я]—\\s[а-я]",
+
+  # 8. Повод —первое упоминание
+  "[а-я]\\s—[а-я]",
+
+  # 9. с разрывом связей С—С и образованием
+  "[А-Я]—[А-Я]",
+
+  # 10. Words that are similar to ordinary, but cannot be at the end of a sentence,
+  #     which means they are abbreviations
+  "\\s(ор|ок|ом|ум|те|ил)\\.",
+
+  # 11. по учению св.отцов
+  "[а-я]\\.[а-я]",
+
+  # 12. др.-евр. чл.-корр.
+  "[а-я]{1,3}\\.-[а-я]{1,4}\\.",
+
+  # 13. ул.Тюменская
+  "[А-Яа-я]\\.[А-Яа-я]",
+
+  # 14. Ставропольский кр.,
+  "\\.,",
+
+  # It is not possible to disallow ALL bad symbols, let's use regex for this
+  # 15. Remove any other bad symbols
+  "[^\\dА-Яа-яёЁ\\s:,.-‑?;!—­‐–'·=’―−”‘]"
```

Comment:
I'd say this disallows a lot of the symbols of `disallowed_symbols`, can you optimize so we only run as many checks as needed? From a very quick glance I'd say we could have `disallowed_symbols` as empty vector, and use this regex for most disallowed symbols. And then together with `even_symbols` it should be fine. I might have missed something though. (see also my other comment)